### ISSUE TITLE
bsRecord on inline fragment

### ISSUE THREAD

**Issue by tmattio** [mhallin/graphql_ppx#60]
Hi,

I am trying to write a query containing union types. I first wrote a separated fragment and got a clear message: only inline fragments are supported with union types. So I wrote my query like this:

```
module Search = [%graphql
  {|
    query Search($query: String!, $searchType: SearchType!) {
      search(first: 10, query: $query, searchType: $searchType) @bsRecord {
        edges @bsRecord {
          node {
            ... on Dataset @bsRecord {
              id
              dataType
            }
          }
        }
      }
    }
  |}
];
```

But the bsRecord on the `... on Dataset @bsRecord` does not work: the fragment is parsed as an object instead of a record. Is this a bug or am I doing something wrong?

Thanks :)

**Comment by anmonteiro** [comment]
Don't you also need `@bsRecord` on `node`?

**Comment by tmattio** [comment]
Adding or removing `@bsRecord` from `node` does not change the parsing either.

**Comment by mhallin** [comment]
This is a bug in the current limitation, thanks for reporting this.

**Comment by mhallin** [comment]
Fixed in 0.2.8.

**Comment by tmattio** [comment]
Hi @mhallin! The bug seems to be fixed in 0.2.8 for union types, but I still have the bug for interfaces :/
